api/app/models/spree/api/webhooks/stock_movement_decorator.rb

```diff
@@ -2,17 +2,30 @@ module Spree
   module Api
     module Webhooks
       module StockMovementDecorator
+        def self.prepended(base)
+          base.around_save :queue_webhooks_requests_for_variant_back_in_stock!
+        end
+
+        # [TODO]: update, this remains as it was
         def update_stock_item_quantity
-          variant_in_stock_before_update = variant_in_stock?
+          variant_in_stock_before_update = stock_item.variant.in_stock?
           super
           variant = stock_item.variant
-          variant.queue_webhooks_requests!('variant.out_of_stock') if variant_in_stock_before_update && !variant_in_stock?
+          variant.queue_webhooks_requests!('variant.out_of_stock') if variant_in_stock_before_update && !stock_item.variant.in_stock?
```

Comment:
@sebastian-palma I'm afraid that at this point of time `in_stock?` will return data from cache - so as I recommended at the beginning this check need to be run on the DB

```ruby
Spree::Variant.in_stock.where(id: variant_id).exists?
```

Reply:
I think the [scope](https://github.com/spree/spree/blob/main/core/app/models/spree/variant.rb#L68) wouldn't be cached, which is in fact cached is the [`in_stock?`](https://github.com/spree/spree/blob/main/core/app/models/spree/variant.rb#L254:L262) instance method. Is that the case @damianlegawiec?

Reply:
@sebastian-palma that's what I wrote - we need to use the SQL query with `in_stock` scope, not `variant.in_stock?`

Reply:
Ah I got it, those changes are for the `variant.out_of_stock` event, that remains untouched in this PR. The purpose of this PR is to add `variant.back_in_stock` and `variant.backorderable` because `variant.out_of_stock` was rejected from QA. Do you want to add those changes to this one @damianlegawiec?

Reply:
`variant.out_of_stock` is updated in [SD-1544](https://github.com/spree/spree/pull/11486/commits/58bf05dc8d01e6b84005feb4423bf9bccc3edbb6) (`full_in_stock?` will be then updated to `in_stock_or_backorderable?`).